### PR TITLE
fix(fileio): improve performances

### PR DESCRIPTION
**Pull request by pbailly**
Cache a classloader.
Improve code complexity of a method.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/533/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/533/)

**Comment by sgandon** [comment]
Hi Pierre, I don't understand this change, the file is a **.java.mine** file, so it does nothing ????

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/541/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/541/)

File: components-api/src/main/java/org/talend/components/api/component/runtime/DependenciesReader.java [hunk 1]
```diff
@@ -169,23 +171,20 @@ public static String computeDependenciesFilePath(String mavenGroupId, String mav
...
-        String s = dependencyString.trim();
-        int indexOfGpSeparator = s.indexOf(':');
-        String groupId = s.substring(0, indexOfGpSeparator);
-        int indexOfArtIdSep = s.indexOf(':', indexOfGpSeparator + 1);
-        String artifactId = s.substring(indexOfGpSeparator + 1, indexOfArtIdSep);
-        int indexOfTypeSep = s.indexOf(':', indexOfArtIdSep + 1);
-        String type = s.substring(indexOfArtIdSep + 1, indexOfTypeSep);
-        int lastIndex = indexOfTypeSep;
+        String trimedDependency = dependencyString.trim();
+        String[] splitedDependency = trimedDependency.split(":");
```
Comment (sgandon) [inline components-api/src/main/java/org/talend/components/api/component/runtime/DependenciesReader.java]: nice, I wonder why I did not do that first.

File: components-api/src/main/java/org/talend/components/api/component/runtime/DependenciesReader.java [hunk 2]
```diff
@@ -73,7 +75,7 @@ public DependenciesReader(String depTxtPath) {
     public Set<String> getDependencies(ClassLoader classLoader) throws IOException {
         ClassLoader zeClassLoader = classLoader;
         if (zeClassLoader == null) {
-            zeClassLoader = this.getClass().getClassLoader();
+            zeClassLoader = this.classLoader;
```
Comment (sgandon) [inline components-api/src/main/java/org/talend/components/api/component/runtime/DependenciesReader.java]: mouaip, I am not sure this provides any improvement.

**Pull request merged**